File: src/npc.cpp
#include <cmath>
#include <sstream>
#include "SDL.h"
#include "../externals/json/json.h"
#include "rect.h"
#include "vectors.h"
#include "math.h"
#include "physics.h"
#include "window.h"
#include "image.h"
#include "npc.h"

#include <iostream>

Npc::Npc(){}
Npc::~Npc(){}
void Npc::Update(){
	Entity::Update();
}
void Npc::Move(float deltaT){
	mPhysics.Move(deltaT);
	Entity::Move(deltaT);
}
void Npc::Draw(Camera *cam){
	Rectf pos = mPhysics.Box();
	if (cam != nullptr)
		pos = Math::FromSceneSpace(cam, pos);

	Window::Draw(&mImage, pos);
}
void Npc::SetMove(int move){
	if (move == Math::LEFT || move == Math::RIGHT)
		mPhysics.SetHorizDir(move);
	else if (move == Math::UP || move == Math::DOWN)
		mPhysics.SetVertDir(move);
	else if (move == Physics::MOVE::STOP){
		mPhysics.SetHorizDir(move);
		mPhysics.SetVertDir(move);
	}
}
Json::Value Npc::Save(){
	Json::Value val = Entity::Save();
	val["obj"]     = "npc";
	val["type"]    = "0";

	return val;
}
void Npc::Load(Json::Value val){
	Entity::Load(val);
}

File: res/scripts/npc.lua
require "Entity"
require "Physics"
require "Rectf"
require "Vector2f"
require "Camera"
require "Color"
require "Timer"
require "Image"
require "Window"
require "Math"
require "State"
require "AnimatedImage"
require "Text"
require "Input"
require "Debug"
require "scripts/luaclass/entity.lua"

npc = class(Entity)

--Init the script
function npc:Init()
	self.img = AnimatedImage("../res/img/animtest.png")
	self.img:play("idle")
	self.physics = self.entity:physics()
end

function npc:Move(deltaT)
	--local speed = 200
	--rect.pos.x = rect.pos.x + LPC.Input.GetJoyAxis(0) * speed * deltaT
	--rect.pos.y = rect.pos.y + LPC.Input.GetJoyAxis(1) * speed * deltaT
	--This gives some odd behaviors
	local v = Vector2f(0, 0)
	v.x = Input.joyAxis(0)
	v.y = Input.joyAxis(1)
	--Set direction isn't implemented
	--physics:setDirection(v)
	--physics:move(deltaT)
end

function npc:Draw(camera)
	--With the changes made in the math class to std::weak_ptr<Camera>
	--We can no longer use the luabind library.
	--LPC.Window.Draw(img, LPC.Math.FromSceneSpace(camera, physics:Box()))
	Window.draw(self.img, Math.fromSceneSpace(camera, self.physics:box()))
end
